dlf/plugins/pageview/tx_dlf_pageview.js

```diff
@@ -396,4 +444,215 @@ dlfViewer.prototype.initLayer = function(imageSourceObjs, isCorsEnabled) {
+}
+
+/**
+ * activate crop selection and set crop values to form
+ */
+dlfViewer.prototype.activateSelection = function () {
+	this.resetCropSelection();
+    this.map.addInteraction(this.draw);
+    
+}
+
+dlfViewer.prototype.initCropping = function () {
+	var viewerObject = this;
+
+    var source = new ol.source.Vector({wrapX: false});
+
+    this.selectionLayer = new ol.layer.Vector({
+    	source: source
+    });
+
+    value = 'LineString';
+    maxPoints = 2;
+    geometryFunction = function(coordinates, geometry) {
+    	if (!geometry) {
+    		geometry = new ol.geom.Polygon(null);
+    	}
+    	var start = coordinates[0];
+      	var end = coordinates[1];
+      	geometry.setCoordinates([
+        	[start, [start[0], end[1]], end, [end[0], start[1]], start]
+      	]);
+
+      	// add to basket button
+        var extent = geometry.getExtent();
+        var imageExtent = viewerObject.map.getLayers().item(0).getExtent();
+
+        var pixel = ol.extent.getIntersection(imageExtent, extent);
+        var rotation = viewerObject.map.getView().getRotation();
+
+        // fill form with coordinates
+        $('#addToBasketForm #startX').val(Math.round(pixel[0]));
+        $('#addToBasketForm #startY').val(Math.round(pixel[1]));
+        $('#addToBasketForm #endX').val(Math.round(pixel[2]));
+        $('#addToBasketForm #endY').val(Math.round(pixel[3]));
+        $('#addToBasketForm #rotation').val(Math.round(viewerObject.radianToDegree(rotation)));
+        //
+
+      	return geometry;
+    };
+
+    this.setNewCropDrawer(source);
+}
+
+/**
+ * reset crop selection
+ */
+dlfViewer.prototype.resetCropSelection = function () {
+	viewerObject = this;
+    this.map.removeLayer(this.selectionLayer);
+    this.source.clear();
+
+	var source = new ol.source.Vector({wrapX: false});
+
+	this.setNewCropDrawer(source);
+
+    this.selectionLayer = new ol.layer.Vector({
+    	source: source
+    });
+
+    this.map.addLayer(this.selectionLayer);
+    
+}
+
+dlfViewer.prototype.setNewCropDrawer = function  (source) {
+	viewerObject = this;
+	this.draw = new ol.interaction.Draw({
+      	source: source,
+      	type: /** @type {ol.geom.GeometryType} */ (value),
+      	geometryFunction: geometryFunction,
+      	maxPoints: maxPoints
+    });
+
+	// reset crop interaction
+    this.draw.on('drawend', function (event) {
+    	viewerObject.map.removeInteraction(viewerObject.draw);
+    });
+
+    this.selectionLayer = new ol.layer.Vector({
+    	source: source
+    });
+}
+
+/**
+ * add magnifier map
+ */
+dlfViewer.prototype.addMagnifier = function (rotation) {
+
+    //magnifier map
+    var extent = [0, 0, 1000, 1000];
+
+    layerProj = new ol.proj.Projection({
+        code: 'goobi-image',
+        units: 'pixels',
+        extent: extent
+    })
+
+    this.ov_view = new ol.View({
+        projection: layerProj,
+        center: ol.extent.getCenter(extent),
+        zoom: 3,
+        rotation: rotation,
+    });
+
+    this.ov_map = new ol.Map({
```

Comment:
It needs an own map object, because the zoom level between origin map and the magnifier map should be different as well as the magnifier map should not have any controls.